### ISSUE TITLE
Name of functions for calculating comps by subareas/areas/regions

### ISSUE THREAD

**Issue by EmilyMarkowitz-NOAA** [afsc-gap-products/gapindex#53]
### Question

I was looking at the `gapindex` R [functions](https://afsc-gap-products.github.io/gapindex/reference/index.html) for calculating comps by areas and noticed that the larger area calculations for sizecomps and agecomps are different. Is there a reason that one is named "[_subarea](https://afsc-gap-products.github.io/gapindex/reference/calc_sizecomp_subarea.html)" and the other "[_region](https://afsc-gap-products.github.io/gapindex/reference/calc_agecomp_region.html)"? Wouldn't they do the same thing? I would advocate for "area" for both, but am also surprised these couldn't be combined with the "*_stratum" functions. More a comment than anything, but it would be good to share naming styles across function names. 

![image](https://github.com/afsc-gap-products/gapindex/assets/75965120/91c1bb33-bd27-4be4-b367-13c06217b92d)

**Comment by zoyafuso-NOAA** [comment]
Yes, the reason that I did not use the term "subarea" when naming the gapindex::calc_agecomp_region() function is that in recreating the historical agecomp tables, the AI/GOA tables provided agecomps only at the region level and the BS tables supported agecomps only at the region and stratum levels. In both regions, agecomps for the other flavors of subareas (e.g., INPFC area, BS subareas, etc.) are not provided. So the only area that stratum-level agecomps are being aggregated over for GAP_PRODUCTS.AGECOMP is at the region level. This naming convention should also make it clear that the function only aggregates and returns region-level agecomps. 

I don't forsee this, but if there is a need for agecomp aggregations at the subarea* level, then I will rework the gapindex::calc_agecomp_region() to do that, change the name to gapindex::calc_agecomp_subarea(), and deprecate the gapindex::calc_agecomp_region() function.  

The *_stratum() functions are separate process from the *_subarea() functions in that the *_stratum() functions do all the heavy lifting with the calculations while the *_subarea() functions are just adding the appropriate stratum estimates together for the various subareas/regions. I saw those as two separate and modular processes. I see your point but this was an easier way for me to develop and troubleshoot the package.
